llvm/lib/Target/AMDGPU/SIISelLowering.cpp

```diff
@@ -4023,10 +4024,20 @@ SDValue SITargetLowering::lowerDYNAMIC_STACKALLOCImpl(SDValue Op,
   Chain = DAG.getCALLSEQ_START(Chain, 0, 0, dl);
```

Comment:
Not related to this PR, but the pair of CALLSEQ_START/CALLSEQ_END is likely redundant.

Reply:
This is also used in the default implementation. 
All the DAG nodes emitted to modify the stack pointer do have a chain dependency and ideally the scheduler should not insert any other stack pointer manipulation between them. I assume the CALLSEQ_START/CALLSEQ_END is a sort of "additional precaution". 
However, should it be removed? I'm not sure. 
@arsenm

Reply:
those are more for the benefit of MI code after the DAG than the chain dependency